.github/workflows/checks.yml

```diff
@@ -0,0 +1,70 @@
+name: CI
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+
+permissions:
+  # Permission for checking out code
+  contents: read
+
+jobs:
+  build:
+    name: Check & Build Provider

```

Comment:
Should we run a `go generate` in the check and build step, to ensure there are no differences during generate?

Reply:
https://github.com/CiscoDevNet/terraform-provider-aci/pull/1203#discussion_r1577223549
Yes, this has already been a WIP. However, just need to fix some issues with the generator running in the CI